spec/abstractops.html

```diff
@@ -1745,22 +1745,29 @@ <h1>
         It parses the argument as either a time zone identifier or an ISO 8601 string.
         The returned Record's fields are set as follows:
         <ul>
-          <li>If _timeZoneString_ is either a named time zone identifier or offset time zone identifier, then [[Name]] is _timeZoneString_, while [[Z]] is *false* and [[OffsetString]] is *undefined*.</li>
-          <li>Otherwise, if _timeZoneString_ is an ISO 8601 string with a time zone annotation containing either a named time zone identifier or offset time zone identifier, then [[Name]] is the time zone identifier contained in the annotation, while [[Z]] is *false* and [[OffsetString]] is *undefined*.</li>
-          <li>Otherwise, if _timeZoneString_ is an ISO 8601 string using a *Z* offset designator, then [[Z]] is *true*, while [[Name]] and [[OffsetString]] are *undefined*.</li>
-          <li>Otherwise, if _timeZoneString_ is an ISO 8601 string using a numeric UTC offset, then [[OffsetString]] is _timeZoneString_, while [[Z]] is *false* and [[Name]] is *undefined*.</li>
+          <li>If _timeZoneString_ is a named time zone identifier, then [[Name]] is _timeZoneString_ and [[OffsetMinutes]] is ~empty~.</li>
+          <li>Otherwise, if _timeZoneString_ is an offset time zone identifier, then [[OffsetMinutes]] is a signed integer and [[Name]] is ~empty~.</li>
+          <li>Otherwise, if _timeZoneString_ is an ISO 8601 string with a time zone annotation containing a named time zone identifier, then [[Name]] is the time zone identifier contained in the annotation and [[OffsetMinutes]] is ~empty~.</li>
+          <li>Otherwise, if _timeZoneString_ is an ISO 8601 string with a time zone annotation containing an offset time zone identifier, then [[OffsetMinutes]] is a signed integer and [[Name]] is ~empty~.</li>
+          <li>Otherwise, if _timeZoneString_ is an ISO 8601 string using a *Z* offset designator, then [[Name]] is *"UTC"* and [[OffsetMinutes]] is ~empty~.</li>
+          <li>Otherwise, if _timeZoneString_ is an ISO 8601 string using a numeric UTC offset, then [[OffsetMinutes]] is a signed integer and [[Name]] is ~empty~.</li>
           <li>Otherwise, a *RangeError* is thrown.</li>
         </ul>
       </dd>
     </dl>
     <emu-alg>
       1. Let _parseResult_ be ParseText(StringToCodePoints(_timeZoneString_), |TimeZoneIdentifier|).
       1. If _parseResult_ is a Parse Node, then
-        1. Return the Record { [[Z]]: *false*, [[OffsetString]]: *undefined*, [[Name]]: _timeZoneString_ }.
+        1. Return ! ParseTimeZoneIdentifier(_timeZoneString_).
```

Comment:
Same concern here as a few lines below: what's the advantage of calling `ParseTimeZoneIdentifier` here, instead of just returning the `{ [[Name]]: _timeZoneString_, [[OffsetMinutes]]: ~empty~ }` record that it will return?

Reply:
It's necessary to call `ParseTimeZoneIdentifier` for the `TimeZoneIdentifier : TimeZoneUTCOffsetName` case to correctly compute `[[OffsetMinutes]]`.

Reply:
Makes sense. I forgot that this PR changes the meaning of [[Name]] in the resulting record from ParseISODateTime's "thing that might be a named ID and might be an offset time zone ID" to ParseTimeZoneIdentifier's "thing that's definitely a named ID". 

This makes me think that at some point we should probably rename [[TimeZone]].[[Name]] in ParseISODateTime, e.g. [[TimeZone]].[[TimeZoneAnnotation]], to avoid other people making the same mistake.

Reply:
> This makes me think that at some point we should probably rename [[TimeZone]].[[Name]] in ParseISODateTime, e.g. [[TimeZone]].[[TimeZoneAnnotation]], to avoid other people making the same mistake.

I opened #2642 to capture this idea